src/target/riscv/riscv.c

```diff
@@ -2696,6 +2696,19 @@ static int riscv_run_algorithm(struct target *target, int num_mem_params,
 		}
 	}
 
+
+	// Disable Interrupts before attempting to run the algorithm.
+	uint64_t current_mstatus;
+	uint8_t mstatus_bytes[8];
+
+	LOG_DEBUG("Disabling Interrupts");
+	register_get(&target->reg_cache->reg_list[REG_MSTATUS]);
+	current_mstatus = buf_get_u64(target->reg_cache->reg_list[REG_MSTATUS].value, 0, info->xlen);
+	buf_set_u64(mstatus_bytes, 0, info->xlen, (current_mstatus & (~((uint64_t)MSTATUS_MIE))) );
```

Comment:
I realize I got the argument order wrong in my previous comment. Sorry.

`set_field(current_mstatus, MSTATUS_MIE, 0)` reads better than the typecast you had to do.
Is it necessary to disable user etc. interrupts as well?